src/access/acp.ts

```diff
@@ -439,3 +484,395 @@ export function internal_getAgentAccessAll(
+      append: false,
+      write: newControlWriteAccess === false,
+    });
+    newAcrPolicy = addIri(newAcrPolicy, acp.allOf, newRule);
+    updatedAcr = setThing(updatedAcr, newAcrPolicy);
+    updatedAcr = setThing(updatedAcr, newRule);
+    newAcrPolicyUrls.push(newAcrPolicyIri);
+    // If we don't have to set new access, we only need to unapply the
+    // ACR Policies that applied to both the given actor and other actors
+    // (because they have been replaced by clones not mentioning the given
+    // actor). Hence `policiesToUnApply` is initialied to `otherActorPolicies`.
+    // However, if we're in this if branch, that means we also had to replace
+    // Policies that defined access for just this actor, so we'll have to remove
+    // all Policies mentioning this actor:
+    acrPoliciesToUnapply = applicableAcrPolicies;
+  }
+
+  const newReadAccess = access.read ?? existingAccess.read;
+  const newAppendAccess = access.append ?? existingAccess.append;
+  const newWriteAccess = access.write ?? existingAccess.write;
+  let policiesToUnapply = otherActorPolicies;
+  if (
+    (typeof newReadAccess !== "undefined" ||
+      typeof newAppendAccess !== "undefined" ||
+      typeof newWriteAccess !== "undefined") &&
+    (newReadAccess !== existingAccess.read ||
+      newAppendAccess !== existingAccess.append ||
+      newWriteAccess !== existingAccess.write)
+  ) {
+    const newPolicyIri =
+      getSourceIri(acr) +
+      `#policy` +
+      `_${encodeURIComponent(actorRelation)}_${encodeURIComponent(actor)}` +
+      `_${Date.now()}_${Math.random()}`;
+    let newPolicy = createPolicy(newPolicyIri);
+    newPolicy = setAllowModes(newPolicy, {
+      read: newReadAccess === true,
+      append: newAppendAccess === true,
+      write: newWriteAccess === true,
+    });
+    newPolicy = setDenyModes(newPolicy, {
+      read: newReadAccess === false,
+      append: newAppendAccess === false,
+      write: newWriteAccess === false,
+    });
+    newPolicy = addIri(newPolicy, acp.allOf, newRule);
+    updatedAcr = setThing(updatedAcr, newPolicy);
+    updatedAcr = setThing(updatedAcr, newRule);
+    newPolicyUrls.push(newPolicyIri);
+    // If we don't have to set new access, we only need to unapply the
+    // Policies that applied to both the given actor and other actors (because
+    // they have been replaced by clones not mentioning the given actor). Hence
+    // `policiesToUnApply` is initialied to `otherActorPolicies`.
+    // However, if we're in this if branch, that means we also had to replace
+    // Policies that defined access for just this actor, so we'll have to remove
+    // all Policies mentioning this actor:
+    policiesToUnapply = applicablePolicies;
+  }
+
+  // ...then remove existing Policy URLs that mentioned both the given actor
+  // and other actors from the given Resource and apply the new ones (but do not
+  // remove the actual old Policies - they might still apply to other Resources!).
```

Comment:
Just to check I'm following, this would be the case if another ACR referenced a Policy in the current ACR for instance ?

Reply:
Yes!